File: hibernate-criteria-builder/src/main/java/io/github/robertomike/hefesto/constructors/ConstructJoinFetch.java
package io.github.robertomike.hefesto.constructors;

import io.github.robertomike.hefesto.actions.JoinFetch;
import jakarta.persistence.criteria.Root;

public class ConstructJoinFetch extends Construct<JoinFetch> {
    public void construct(Root<?> root) {
        items.forEach(value -> root.fetch(value.getRelationship(), value.getJoinType()));
    }
}

File: hibernate-criteria-builder/src/main/kotlin/io/github/robertomike/hefesto/constructors/ConstructJoinFetch.kt
package io.github.robertomike.hefesto.constructors

import io.github.robertomike.hefesto.actions.JoinFetch
import jakarta.persistence.criteria.Root

class ConstructJoinFetch : Construct<JoinFetch>() {
    fun construct(root: Root<*>) {
        items.forEach { value ->
            root.fetch<Any, Any>(value.relationship, value.joinType)
        }
    }
}
